File: program/convert-hexadecimal-to-decimal/ConvertHexadecimalToDecimal.ts
const alphabetToNumberMap = {
  a: 10,
  b: 11,
  c: 12,
  d: 13,
  e: 14,
  f: 15,
};

const convertHexadecimalToDecimal = (hexaDecimal: string): number => {
  let sum = 0;

  for (let i = hexaDecimal.length - 1; i >= 0; i--) {
    const value =
      alphabetToNumberMap[hexaDecimal[i].toLowerCase()] ||
      Number(hexaDecimal[i]);

    sum += value * Math.pow(16, hexaDecimal.length - 1 - i);
  }

  return sum;
};

console.log(convertHexadecimalToDecimal("a"));

File: program/program/convert-hexadecimal-to-decimal/ConvertHexadecimalToDecimal.kt
import kotlin.math.pow

fun main() {
    print("Enter Hexadecimal Number to Convert: ")
    val num = readln()
    print("Decimal Number is: ${convert(num)}")
}

private fun convert(str: String): Int {
    var num = 0
    var i = str.lastIndex
    str.forEach {
        val digit = when (it) {
            'A' -> 10
            'B' -> 11
            'C' -> 12
            'D' -> 13
            'E' -> 14
            'F' -> 15
            else -> it - '0'
        }
        num += 16.0.pow(i).toInt() * digit
        i--
    }
    return num
}